// fake-users-array/index.js
var users = [
    {
                name : 'Frank',
                surname : 'Castle',
                dateOfBirth : '15-12-1969',
                email : 'frankcastle@gmail.com',
                id : 1
    },
    {
                name : 'Matt',
                surname : 'Murdock',
                dateOfBirth : '02-12-1968',
                email : 'mattmurdock@gmail.com',
                id : 2
    }];
var resetUsers = [];
Object.assign(resetUsers, users);

exports.addUser = function(name,surname,dateOfBirth,email){
    users.push({
                name : name,
                surname : surname,
                dateOfBirth : dateOfBirth,
                email : email,
                id : users[users.length-1].id+1
    });

    return users;
}

exports.getUsers = function() {
  return users;
}

exports.getUsersById = function(id) {
  for (var i = 0; i < users.length; i++) {
    if (id==users[i].id) {
      return users[i];
    }
  }
  return null;
}

exports.deleteUserById = function(id){
  for (var i = 0; i < users.length; i++) {
    if (id==users[i].id) {
      users.splice(i,1);
    }
  }
  return users;
}

exports.reset = function() {
  users = resetUsers;
  return resetUsers;
}

exports.modifyUserById = function(id, obj){
  for (var i = 0; i < users.length; i++) {
    if (id==users[i].id) {
      if (obj.name!==undefined) {
        users[i].name = obj.name;
      }

      if (obj.surname!==undefined) {
        users[i].surname = obj.surname;
      }

      if (obj.dateOfBirth!==undefined) {
        users[i].dateOfBirth = obj.dateOfBirth;
      }

      if (obj.email!==undefined) {
        users[i].email = obj.email;
      }
    }
  }
  return users;
}
